collect_app/src/main/java/org/odk/collect/android/tasks/FormLoaderTask.java

```diff
@@ -116,80 +116,31 @@ public FormLoaderTask(String instancePath, String xpath, String waitingXPath) {
 
     /**
      * Initialize {@link FormEntryController} with {@link FormDef} from binary or
-     * from XML. If given an instance, it will be used to fill the {@link FormDef}
-     * .
+     * from XML. If given an instance, it will be used to fill the {@link FormDef}.
      */
     @Override
     protected FECWrapper doInBackground(String... path) {
-        FormDef fd = null;
-        FileInputStream fis = null;
         errorMsg = null;
 
-        String formPath = path[0];
-
-        File formXml = new File(formPath);
-        String formHash = FileUtils.getMd5Hash(formXml);
-        File formBin = new File(Collect.CACHE_PATH + File.separator + formHash + ".formdef");
-
-        publishProgress(
-                Collect.getInstance().getString(R.string.survey_loading_reading_form_message));
+        final String formPath = path[0];
+        final File formXml = new File(formPath);
 
+        final FormDef formDef = createFormDefFromCacheOrXml(formPath, formXml);
 
-        //    FormDef.setDefaultEventNotifier(new EventNotifier() {
```

Comment:
Verified that the deleted block below was properly converted into `createFormDefFromCacheOrXml`.